R/readGAINS2025final.R

```diff
@@ -0,0 +1,285 @@
+      )
+
+    # 1) Filtering for one scenario (not VLLO) per SSP, removing the others, and calling it "SMIPbySSP"
+    smipefsSMIPbySSP <- smipefs %>%
+      filter((ssp %in% c("SSP1", "SSP2", "SSP3") & scenario == "Medium") | (ssp == "SSP5" & scenario == "High")) %>%
+      mutate(scenario = "SMIPbySSP")
+    # 2) Filtering for VLLO and SSP1, removing the others, and calling it SMIPVLLO
+    smipefsSMIPVLLO <- smipefs %>%
+      filter(ssp == "SSP1" & scenario == "Very Low Limited Overshot") %>%
+      mutate(ssp = "SMIPVLLO", scenario = "SMIPVLLO")
+    # Combining the filtered dataframes
+    smipefs <- rbind(smipefsSMIPbySSP, smipefsSMIPVLLO)
+    rm(smipefsSMIPbySSP, smipefsSMIPVLLO)
+
+    # Maximum Technically Feasible Reduction (MTFR) ==================================================
+    # No corrections for Transformations_Coal and Transformations_NatGas
+    mtfrefs <- readSectors("SSPs_IMAGE_emf", "mtfr_2025-07-02.csv")
+
+    # Convert to long format and standardize column names
+    mtfrefs <- pivot_longer(mtfrefs, 5:length(names(mtfrefs)), names_prefix = "X", names_to = "year") %>%
+      rename(
+        scenario = SCENARIO,
+        region = Group_Region,
+        sectorGAINS = EMF30_MIXED,
+        species = POLLUTANT_FRACTION
+      ) %>%
+      mutate(
+        ssp = "MTFR",
+        scenario = "MTFR"
+      )
+
+    # Strong Legislation Emissions (SLE) =============================================================
+    # No corrections for Transformations_Coal and Transformations_NatGas
+    sleefs <- readSectors("SSPs_IMAGE_emf", "middle_2025-07-02.csv")
+
+    # Convert to long format and standardize column names
+    sleefs <- pivot_longer(sleefs, 5:length(names(sleefs)), names_prefix = "X", names_to = "year") %>%
+      rename(
+        ssp = scen,
+        region = Group_Region,
+        sectorGAINS = EMF30_MIXED,
+        species = POLLUTANT_FRACTION
+      ) %>%
+      mutate(
+        scenario = "SLE"
+      )
+
+    # Current Legislation Emissions (CLE) =============================================================
+    # No corrections for Transformations_Coal and Transformations_NatGas
+    # Also contains historical emission factors. Remove ???
+    cleefs <- readSectors("SSPs_IMAGE_emf", "cle_rev_2025-07-02.csv")
+
+    # Convert to long format and standardize column names
+    cleefs <- pivot_longer(cleefs, 5:length(names(cleefs)), names_prefix = "X", names_to = "year") %>%
+      rename(
+        ssp = scen,
+        region = Group_Region,
+        sectorGAINS = EMF30_MIXED,
+        species = POLLUTANT_FRACTION
+      ) %>%
+      mutate(
+        scenario = "CLE"
+      )
+
+    # Remove historical emissions factors and historical years.
+    # These are read in from a different source file for scenario "historical".
+    cleefs <- cleefs %>% filter(ssp != "historical", year >= 2025)
+
+    # COMPLETE DATA FRAMES TO ENSURE THAT ALL SECTOR x SPECIES x REGION COMBINATIONS ARE PRESENT =====
```

Comment:
Are you sure that this step is necessary (i.e. all the complete statements)? The conversion to magclass should work without it and it would fill the gaps itself.